### ISSUE TITLE
Can this plugin be used to generate id for sub documents?

### ISSUE THREAD

**Issue by tahmidsadik112** [typegoose/auto-increment#3]

**Comment by hasezoey** [comment]
as long as mongoose executes the hooks & plugins (what it *should* do), yes

**Comment by tahmidsadik112** [comment]
https://github.com/typegoose/auto-increment/blob/master/src/autoIncrement.ts#L100
when trying to use auto increment plugin on a sub-document 
`this.db` is undefined on this line https://github.com/typegoose/auto-increment/blob/master/src/autoIncrement.ts#L100

This is how the plugin is being used
```
@plugin<AutoIncrementIDOptions>(AutoIncrementID, {
  field: 'id',
  startAt: 1,
  incrementBy: 1
})
@modelOptions({
  schemaOptions: {
    timestamps: true
  }
})
export class Loan extends Base { 
constructor() {
}
@prop()
id!: number;
......
 } 
```

**Comment by hasezoey** [comment]
i will look into this, could you provide some information?
- nodejs version
- do you use it with typegoose (i assume yes - from your example)
- typescript version
- mongoose version

**Comment by tahmidsadik112** [comment]
* Node version - `12.10.0`
* Do you use it with typegoose - `Yes`
* typescript version - `3.8.2`
* mongoose version - `5.9.3`

**Comment by hasezoey** [comment]
because i couldnt find an way to access the connection of the model used in an subdocument applied plugin, i made [an new issue](https://github.com/Automattic/mongoose/issues/8646) at mongoose's repository

**Comment by hasezoey** [comment]
thanks to [this comment](https://github.com/Automattic/mongoose/issues/8646#issuecomment-599257442) i could resolved the issue you had, released in 0.4.0

**Comment by tahmidsadik112** [comment]
Tested and works as expected, Thanks :+1: :heart:
